src/state_machine/coordinator/mod.rs

```diff
@@ -1117,19 +1117,23 @@ pub mod test {
             }
         }
 
+        assert_eq!(outbound_messages.len(), 1);
+        let messages = outbound_messages.clone();
+        let result = feedback_messages_with_errors(&mut coordinators, &mut signers, &messages);
+        assert!(result.is_ok());
+
         // add NonceResponses so there's more than the number of signers
         let mut packet = outbound_messages[0].clone();
         if let Message::SignatureShareRequest(ref mut request) = packet.msg {
-            for _ in 0..1000 {
-                request
-                    .nonce_responses
-                    .push(request.nonce_responses.first().unwrap().clone());
-            }
+            request
```

Comment:
So rather than testing an excessively large number of nonce responses, as in the original bug, you're simply testing a single duplicate?

Reply:
> So rather than testing an excessively large number of nonce responses, as in the original bug, you're simply testing a single duplicate?

Yup. Testing for any duplicates implicitly covers too large a number of nonce responses. The previous testing approach could have allowed for a single duplicate, even though the code was fine.

This test isn't perfect either.

> Should we also try an out-of-range `signer_id`?

Yeah, that's one of the tests that I wanted to add. We need to test for all of the conditions checked for in the code.

Reply:
Ok, I added tests cases with no `NonceResponse` and with and out-of-range `signer_id`.